gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/access/heap/MemHeapScanController.java

```diff
@@ -407,15 +407,16 @@ else if (this.lockPolicy.zeroDurationReadLocks()) {
         restoreBatching = this.localTXState.getProxy().remoteBatching(true);
       }
     }
-    else if (Misc.getGemFireCache().snapshotEnabled()) {
+    else if (this.gfContainer.isRowBuffer() ||
+        (region.getCache().snapshotEnabledForTest() && region.getConcurrencyChecksEnabled())) {
+
       // Start snapshot tx only for read operations.but not for read_only mode
-      boolean forReadOnly = (this.openMode & GfxdConstants
-          .SCAN_OPENMODE_FOR_READONLY_LOCK) != 0;
+      //boolean forReadOnly = (this.openMode & GfxdConstants
+      //    .SCAN_OPENMODE_FOR_READONLY_LOCK) != 0;
       if (region.getConcurrencyChecksEnabled() &&
-          region.getCache().snapshotEnabled() &&
           (region.getCache().getCacheTransactionManager().getTXState() == null)
-          && (this.forUpdate == 0)
-          && !forReadOnly) {
+          /*&& (this.forUpdate == 0)*/
+          /*&& !forReadOnly*/) {
```

Comment:
This is only in case of snapshot. row store/row table doesn't support snapshot isolation.
This was added when I was trying to support snapshot in RC mode. Not needed now.